src/units/unit.cpp

```diff
@@ -1233,6 +1233,11 @@ void unit::end_turn()
 	}
 
 	set_state(STATE_NOT_MOVED,false);
+
+	// Without this, on the other player's turn, the unit will be seen by enemies/observers
+	// as having extra movement range (total_movement() + movement_).
+	movement_ = 0;
```

Comment:
This is a game state change that might cause oos errors and thus cannot go into 1.14 unless it either fixes something really important ( if we for some reason decide to block outdated wesnoth versions on the mp server this would become acceptable as well but currently we are not intending to do that)

Reply:
OK, it's nothing critical for touch interface. Will move to another commit and remove it from PR.